WcaOnRails/app/views/registrations/register.html.erb

```diff
@@ -42,7 +42,7 @@
     <% end %>
   <% else %>
 
-    <% if @competition.registration_opened? %>
+    <% if @competition.registration_opened? && @competition.showAtAll? %>
```

Comment:
I personally think it's good that we allow organizers to register for an invisible competition, although it is understandable that it confused @Luis-J-Ianez. How about we just add an alert that makes it very clear that they're allowed to register for their comp, but other people even with the link won't be able to?

Also, can you create a separate PR for this? I'd rather not have different issues being address in the same PR.